stream_alert/rule_processor/metrics.py

```diff
@@ -0,0 +1,122 @@
+'''
+Copyright 2017-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+import json
+
+from datetime import datetime
+
+import boto3
+
+from botocore.exceptions import ClientError
+
+from stream_alert.rule_processor import LOGGER
+
+
+class Metrics(object):
+    """Class to hold Rule Processor metric name and unit constants
+    This basically acts as an enum, allowing for the use of dot notation for
+    accessing properties and avoids doing dict lookups a ton.
+    """
+
+    def __init__(self, region):
+        self.boto_cloudwatch = boto3.client('cloudwatch', region_name=region)
+
+    class Name(object):
+        """Constant metric names used in the rule processor"""
+        FAILED_PARSES = 'RuleProcessorFailedParses'
+        S3_DOWNLOAD_TIME = 'RuleProcessorS3DownloadTime'
+        TOTAL_RECORDS = 'RuleProcessorTotalRecords'
+        TOTAL_S3_RECORDS = 'RuleProcessorTotalS3Records'
+        TRIGGERED_ALERTS = 'RuleProcessorTriggeredAlerts'
+
+    class Unit(object):
+        """Unit names for metrics. These are taken from the boto3 CloudWatch page"""
+        SECONDS = 'Seconds'
+        MICROSECONDS = 'Microseconds'
+        MILLISECONDS = 'Milliseconds'
+        BYTES = 'Bytes'
+        KILOBYTES = 'Kilobytes'
+        MEGABYTES = 'Megabytes'
+        GIGABYTES = 'Gigabytes'
+        TERABYTES = 'Terabytes'
+        BITS = 'Bits'
+        KILOBITS = 'Kilobits'
+        MEGABITS = 'Megabits'
+        GIGABITS = 'Gigabits'
+        TERABITS = 'Terabits'
+        PERCENT = 'Percent'
+        COUNT = 'Count'
+        BYTES_PER_SECOND = 'Bytes/Second'
+        KILOBYTES_PER_SECOND = 'Kilobytes/Second'
+        MEGABYTES_PER_SECOND = 'Megabytes/Second'
+        GIGABYTES_PER_SECOND = 'Gigabytes/Second'
+        TERABYTES_PER_SECOND = 'Terabytes/Second'
+        BITS_PER_SECOND = 'Bits/Second'
+        KILOBITS_PER_SECOND = 'Kilobits/Second'
+        MEGABITS_PER_SECOND = 'Megabits/Second'
+        GIGABITS_PER_SECOND = 'Gigabits/Second'
+        TERABITS_PER_SECOND = 'Terabits/Second'
+        COUNT_PER_SECOND = 'Count/Second'
+        NONE = 'None'
+
+    def put_metric_data(self, metric_name, value, unit):
```

Comment:
For example, `FailedParses` and `TriggeredAlerts` are sent right after each other; these could be consolidated. 

My strategy was to publish all the metrics at the end, unless there are important metrics that might be lost because an exception was raised somewhere.

Happy to chat offline!